CI: test_window_lifecycle still flaky after #446 fix — Notepad PID not found in window list

## Problem
`test_window_lifecycle` in `tests/test_phase2_comprehensive.py` still fails intermittently after the fix in commit 3a36c4f (#446). The test launches Notepad, gets its PID, but then cannot find that PID in the window list.

## Steps to Reproduce
```bash
python -m pytest tests/test_phase2_comprehensive.py::TestE2EWorkflows::test_window_lifecycle -v --timeout=30
```

## Expected
Test passes — Notepad PID appears in window list.

## Actual
```
AssertionError: Notepad (PID 28816) should appear in window list
assert 0 >= 1
```
Fails consistently in this CI run (both desktop and E2E suites), even though #446 supposedly fixed it by tracking own PID instead of global counts.

## CI Results (commit eb94a93)
- Desktop+integration: 109 passed, **1 failed**, 18 skipped, 2 xpassed
- E2E: 8 passed, **1 failed**, 1 xfailed, 3 xpassed
- Only failure in both suites is this test

## Prior Issues
- #428 (closed): CI: test_window_lifecycle flaky — leftover windows cause count mismatch
- #446 (closed): test_window_lifecycle flaky: orphan Notepad windows accumulate across test runs

## Environment
- Windows 11 Pro 10.0.26200
- naturo 0.3.0
- Runner: NATUROBOT
- 20+ stale Notepad windows from previous sessions

**[QA-Mariana]**

--- a/tests/test_phase2_comprehensive.py
+++ b/tests/test_phase2_comprehensive.py
@@ -786,20 +786,40 @@ class TestE2EWorkflows:
         """T039: Window lifecycle: launch notepad → appears in list → close → disappears."""
         import subprocess
 
-        # (#446) Track the specific window by PID instead of comparing
-        # global window counts — orphan Notepad windows from other tests
-        # caused false failures.
+        # (#472) Record existing Notepad windows BEFORE launch so we can
+        # identify the new one even if orphan windows exist from other tests.
+        pre_hwnds = {
+            w.handle for w in core.list_windows()
+            if "notepad" in w.process_name.lower()
+        }
+
         proc = subprocess.Popen(["notepad.exe"])
         try:
-            time.sleep(1.5)
+            # (#472) Poll for the window to appear instead of a fixed sleep.
+            # On busy CI runners, 1.5s is sometimes not enough.  Also handle
+            # the case where Windows re-launches Notepad under a child PID
+            # by falling back to process-name matching for new HWNDs.
+            our_hwnd = None
+            deadline = time.monotonic() + 10.0
+            while time.monotonic() < deadline:
+                for w in core.list_windows():
+                    if w.pid == proc.pid:
+                        our_hwnd = w.handle
+                        break
+                    # Fallback: new Notepad window not seen before launch
+                    if (
+                        "notepad" in w.process_name.lower()
+                        and w.handle not in pre_hwnds
+                    ):
+                        our_hwnd = w.handle
+                        break
+                if our_hwnd is not None:
+                    break
+                time.sleep(0.3)
 
-            # Verify our Notepad appears (match by PID)
-            our_windows = [
-                w for w in core.list_windows()
-                if w.pid == proc.pid
-            ]
-            assert len(our_windows) >= 1, (
-                f"Notepad (PID {proc.pid}) should appear in window list"
+            assert our_hwnd is not None, (
+                f"Notepad (PID {proc.pid}) should appear in window list "
+                f"within 10s"
             )
 
         finally:
@@ -811,7 +831,7 @@ class TestE2EWorkflows:
             while time.monotonic() < deadline:
                 remaining = [
                     w for w in core.list_windows()
-                    if w.pid == proc.pid
+                    if w.handle == our_hwnd
                 ]
                 if not remaining:
                     break
@@ -819,12 +839,12 @@ class TestE2EWorkflows:
             else:
                 remaining = [
                     w for w in core.list_windows()
-                    if w.pid == proc.pid
+                    if w.handle == our_hwnd
                 ]
 
             assert not remaining, (
-                f"Notepad (PID {proc.pid}) should disappear from list "
-                f"after termination, but {len(remaining)} window(s) remain"
+                f"Notepad window (HWND {our_hwnd}) should disappear "
+                f"after termination, but still visible"
             )
 
     def test_drag_from_a_to_b(self, backend):

